dtls.c

```diff
@@ -3358,17 +3397,18 @@ decrypt_verify(dtls_peer_t *peer, uint8 *packet, size_t length,
   if (security->cipher == TLS_NULL_WITH_NULL_NULL) {
     /* no cipher suite selected */
     return clen;
-  } else { /* TLS_PSK_WITH_AES_128_CCM_8 or TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8 */
+  } else { /* TLS_PSK_WITH_AES_128_CCM_8, TLS_PSK_WITH_AES_128_CCM, TLS_ECDHE_ECDSA_WITH_AES_128_CCM_8 or TLS_ECDHE_ECDSA_WITH_AES_128_CCM */
     /**
      * length of additional_data for the AEAD cipher which consists of
      * seq_num(2+6) + type(1) + version(2) + length(2)
      */
 #define A_DATA_LEN 13
     unsigned char nonce[DTLS_CCM_BLOCKSIZE];
     unsigned char A_DATA[A_DATA_LEN];
+    int macLen = get_cipher_suite_mac_len(security->cipher);
     /* For backwards-compatibility, dtls_encrypt_params is called with
      * M=<macLen> and L=3. */
-    const dtls_ccm_params_t params = { nonce, 8, 3 };
+    const dtls_ccm_params_t params = { nonce, macLen, 3 };
 
     if (clen < 16)		/* need at least IV and MAC */
```

Comment:
I think that now, we might need to have to check for something like `clen < (mac_len + MAX_IV_LENGTH)`, with `MAX_IV_LENGTH` currently being 8. 
Otherwise, the newly added `clen - macLen` below might yield an underflow.

Reply:
done.
(there is no symbolic IV-length, right?)

Reply:
Correct—for some unknown reason, that has not been defined when crafting this check.

Reply:
I would propose to go without for this PR.
And have a second PR introducing [RECORD_IV_LENGTH](https://datatracker.ietf.org/doc/html/rfc5246#page-25). 

WDYT?

Reply:
Sure, a separate PR would be the cleanest  solution.